### PR TITLE
Wip/h3nk3 new template cache

### PR DESCRIPTION
**Pull request by henrikengstrom**

### CODE REVIEW

File: project/build.scala [hunk 1]
```diff
@@ -108,7 +108,6 @@ object TheActivatorBuild extends Build {
     settings(PlayKeys.playDefaultPort := 8888)
     settings(Keys.includeFilter in (Assets, LessKeys.less) := "*.less")
     settings(Keys.excludeFilter in (Assets, LessKeys.less) := "_*.less")
-    settings(Keys.initialize ~= { _ => sys.props("scalac.patmat.analysisBudget") = "512" })
```
Comment (corruptmemory) [inline project/build.scala]: Why is this no-longer needed?  What caused this to be needed before?

Comment (henrikengstrom) [inline project/build.scala]: Seems you added this waaaay back: https://github.com/typesafehub/activator/commit/ce007c6eca60535c334e0aea37e73cc71d27adf5

The reason for not using this anymore is we get tons of this warning in the build right now:
``
[warn] /Users/he/code/typesafe/activator/echo/trace/event/src/main/scala/com/typesafe/trace/util/ProtobufConverter.scala:1187: Please remove -Dscalac.patmat.analysisBudget, it is ignored.
[warn]   private[util] def fromProtoSystemMessage(proto: ProtoAnnotationData): SysMsg = proto.getSysMsg match {
[warn]                     ^
[warn] /Users/he/code/typesafe/activator/echo/trace/event/src/main/scala/com/typesafe/trace/util/ProtobufConverter.scala:1204: Please remove -Dscalac.patmat.analysisBudget, it is ignored.
[warn]   private[util] def fromProtoInfo(info: ProtoInfo): Info = info.getType match {
``

Comment (corruptmemory) [inline project/build.scala]: Interesting, something must have changed in scala.

**Comment by corruptmemory** [comment]
LGTM

File: project/build.scala (hunk 1 above)
Comment (gkossakowski) [inline project/build.scala]: See scala/scala#4078

Comment (henrikengstrom) [inline project/build.scala]: Thanks for the explanation @gkossakowski

**Pull request merged**